code/src/pages/cart/components/CartItem.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from "react";
+
+import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
+import DeleteIcon from "@mui/icons-material/Delete";
+import RemoveCircleOutlineIcon from "@mui/icons-material/RemoveCircleOutline";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import formatPrice from "@/utils/format-price/formatPrice";
+
+import "@/pages/cart/components/CartItem.scss";
+
+type ProductCartType = {
+  image: string;
+  name: string;
+  productPrice: number;
+  quantity: number;
+  calculatedPrice: number;
+};
+
+type CartItemProps = {
+  item: ProductCartType;
+  onQuantityChange: (quantity: number) => void;
+  onDelete: () => void;
+};
+
+const CartItem = ({ item, onQuantityChange, onDelete }: CartItemProps) => {
+  const [quantity, setQuantity] = useState(item.quantity.toString());
+
+  useEffect(() => {
+    setQuantity(item.quantity.toString());
+  }, [item.quantity]);
+
+  const handleDecrement = () => {
+    if (item.quantity > 1) {
+      onQuantityChange(item.quantity - 1);
+    }
+  };
+
+  const handleIncrement = () => {
+    onQuantityChange(item.quantity + 1);
+  };

```

Comment:
Also, I think it would be better to move inside CartSlice

Reply:
> Also, I think it would be better to move inside CartSlice

It will be nice if you read PR's description before review